### PR TITLE
Translate Design Principals

### PR DESCRIPTION
**Pull request by caojiele**
## What is the purpose of the change

#196 Add english version of http://dubbo.apache.org/zh-cn/docs/dev/principals
On the basis of biyuhao, complete translation.

## Brief changelog

- [x] code-detail.md
- [x] configuration.md
- [x] dummy.md
- [x] expansibility.md
- [x] extension.md
- [x] general-knowledge.md
- [x] introduction.md
- [x] robustness.md

Follow this checklist to help us incorporate your contribution quickly and easily:

- [x] Make sure there is a [GITHUB_issue](https://github.com/apache/incubator-dubbo-website/issues) filed for the change (usually before you start working on it). Trivial changes like typos do not require a GITHUB issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Format the pull request title like `Fix UnknownException when host config not exist #XXX`. Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Test your code locally by running `docsite start`, and make sure it works as expected.
- [ ] Make sure no files under build directory is added.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/incubator-dubbo/wiki/Software-donation-guide).

### CODE REVIEW

File: docs/en-us/dev/principals/configuration.md [hunk 1]
```diff
@@ -1,84 +1,83 @@
...
+Forwards compatibility is very good, as long as you guarantee configuration only grow, basically can guarantee forwards compatibility.But backward compatibility, but also should pay attention to, to prepare for subsequent to join the new configuration items.If a special configuration in configuration, you should make an appointment a compatibility for the "special" case rules, because the special case, probably will happen later.For example, have a configuration file is save "service = address mapping", one of the special line, is saved "registry = address".Now application load time to define "registry" the Key is special, do special processing, the other is "service".New version found, however, to add a "monitoring center" = address, at this point, the old version of the program will handle "monitoring center" as a "service", because the old code can't be change, compatibility is will be very troublesome.If previously agreed upon in the "special" logo + XXX for special treatment, follow-up will be more convenient. 
 
 向后兼容性，可以多向HTML5学习，参见：[HTML5设计原理](http://javatar.iteye.com/blog/949390)
```
Comment (kezhenxu94) [inline docs/en-us/dev/principals/configuration.md]: Translate this please

File: docs/en-us/dev/principals/robustness.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+### On the database lock
+
+Registration center will regularly update the database of a record timestamp, such cluster other registry perceive it is alive.Overdue registry and its associated data will be cleared.Database is normal, the mechanism as well.But the database load is high, its every action is slow.This occurs:
+
+A registry that B expired, delete B data.B find their data, to write their own data repeatedly.These repeated operation and increase load the database, deterioration.
+
+Use the following logic:
+
+When data is deleted B found themselves fail (write), choose to wait for this period of time and try again.Can choose to retry time exponentially, such as first class 1 minute, the second for 10 minutes, 100 minutes for the third time.
+这样操作减少后，保证数据库可以冷却（Cool Down）下来。
```
Comment (kezhenxu94) [inline docs/en-us/dev/principals/robustness.md]: Translate this please

File: docs/en-us/dev/principals/expansibility.md [hunk 3]
```diff
@@ -1,18 +1,18 @@
-# 谈谈扩充式扩展与增量式扩展
+# Talk about expansion of extension and incremental extension
 
 > http://javatar.iteye.com/blog/690845
 
 
-我们平台的产品越来越多，产品的功能也越来越多。平台的产品为了适应各 BU 和部门以及产品线的需求，势必会将很多不相干的功能凑在一起，客户可以选择性的使用。为了兼容更多的需求，每个产品，每个框架，都在不停的扩展，而我们经常会选择一些扩展的扩展方式，也就是将新旧功能扩展成一个通用实现。我想讨论是，有些情况下也可以考虑增量式的扩展方式，也就是保留原功能的简单性，新功能独立实现。我最近一直做分布式服务框架的开发，就拿我们项目中的问题开涮吧。 
+Our platform products more and more, the function of the product also more and more.Platform products in order to meet the requirement of each BU and department as well as product line, will surely will be a lot of irrelevant function together, the customer can use selective.In order to compatible with more demand for each product, each framework, are constantly expanding, and we often choose some extension of the extension, namely to old and new function expanded into a general implementation.I want to discuss is, in some cases also can consider to the expansion of the incremental way, also is to retain the original function of simplicity, new feature implementation independence.I have been doing the development of distributed service framework, and then take such problem in our project.
```
Comment (kezhenxu94) [inline docs/en-us/dev/principals/expansibility.md]: There are more and more products in our platform

File: docs/en-us/dev/principals/general-knowledge.md [hunk 4]
```diff
@@ -1,69 +1,68 @@
...
+This is part of the design by contract, try to keep to a return value method is to query methods, void return method is to command.Query methods are usually idempotence, without side effects, also is not change any state, the n results are the same, such as the get a property value, or query a database record.Command is to point to have side effects, namely will change state, such as set a value, or update a database record.If you have modified the status of the operation, do a query returns again, if possible, to split it into writing reading separation of the two methods, such as: User deleteUser (id), delete users and returns the deleted users, consider to the getUser () and void deleteUser ().In addition, each method is front-facing assert that the legitimacy of the incoming parameters, as far as possible the rear assertion returns the legitimacy, and documented. 
 
-## 增量式扩展，而不要扩充原始核心概念
+## Incremental extension, and not to extend the original core concept
 
-参见：[谈谈扩充式扩展与增量式扩展](./principals/expansibility.md)
\ No newline at end of file
+参见：[expansibility](./principals/expansibility.md)
```
Comment (kezhenxu94) [inline docs/en-us/dev/principals/general-knowledge.md]: translate this please

File: docs/en-us/dev/principals/general-knowledge.md [hunk 5]
```diff
@@ -65,4 +65,4 @@ This is part of the design by contract, try to keep to a return value method is
 
 ## Incremental extension, and not to extend the original core concept
 
-参见：[expansibility](./principals/expansibility.md)
+see here：[expansibility](./principals/expansibility.md)
```
Comment (kezhenxu94) [inline docs/en-us/dev/principals/general-knowledge.md]: see here => refer to

File: docs/en-us/dev/principals/robustness.md [hunk 6]
```diff
@@ -1,84 +1,83 @@
...
+Dubbo as a remote service exposure, calls and management solutions, through the meridians of the application is running, its itself to achieve robustness of importance is self-evident.
 
-这里列出一些 Dubbo 用到的原则和方法。
+Here are some Dubbo principle and method of use.
 
-## 日志
+## The log
 
-日志是发现问题、查看问题一个最常用的手段。日志质量往往被忽视，没有日志使用上的明确约定。重视 Log 的使用，提高 Log 的信息浓度。日志过多、过于混乱，会导致有用的信息被淹没。
+Logging is found, view problems one of the most commonly used method.Log quality is often neglected, there is no log on using expressly agreed upon.Attaches great importance to the use of the Log, and improve the concentration of the Log information.Log too much, too much chaos, could lead to useful information.
```
Comment (kezhenxu94) [inline docs/en-us/dev/principals/robustness.md]: Logging is one of the most commonly used way to find, discover problems

**Comment by caojiele** [comment]
> @caojiele please refer to the review comments I left inline, and add a space() after each period(`.`)
> 
> And it seems that there are some grammer errors, others can help to review and give suggestions here.

Well, I've already changed according to your requirements.

**Pull request merged**